torch_audiomentations/augmentations/impulse_response.py

```diff
@@ -74,11 +77,29 @@ def apply_transform(self, selected_samples, sample_rate: int = None):
 
         batch_size, num_channels, num_samples = selected_samples.shape
 
-        # (batch_size, max_ir_length, 1)
+        # (batch_size, 1, max_ir_length)
         ir = self.transform_parameters["ir"].to(selected_samples.device)
 
         convolved_samples = convolve(
             selected_samples, ir.expand(-1, num_channels, -1), mode=self.convolve_mode
         )
 
-        return convolved_samples[..., :num_samples]
+        if self.fix_propagation_delay:
+            propagation_delays = ir.abs().argmax(dim=2, keepdim=False)[:, 0]
+
+            convolved_samples = torch.stack(
+                [
+                    convolved_sample[
+                        :, propagation_delay : propagation_delay + num_samples
+                    ]
+                    for convolved_sample, propagation_delay in zip(
+                        convolved_samples, propagation_delays
+                    )
+                ],
+                dim=0,
+            )
+
+            return convolved_samples
```

Comment:
Why don't we want to do `[..., :num_samples]` here? In my code, I did it to ensure that the output has the same shape as the input, instead of being longer. In other words, I cut off the reverb tail

Reply:
Line 92 can be written as:

```python
convolved_sample[:, propagation_delay:][...,:num_samples]
```

1. We start by getting rid of the first `propagation_delay` with `[..., propagation_delay:]` 
2. We cut off the reverb tail as you suggest with `[..., :num_samples]`

Point 1. is what the `fix_propagation_delay` option is used for.
Or maybe I completely got wrong this propagation delay thing  (that is definitely an option :))

Reply:
Ah, you're right - I didn't read the slice code above in detail. This works as expected already.